eclipse-collections-code-generator/src/test/java/org/eclipse/collections/codegenerator/tools/JavadocUtil.java

```diff
@@ -0,0 +1,206 @@
+            {
+                javadocLines.add(srcLineNorm);
+                if (srcLineTrimmed.startsWith("*/") || srcLineTrimmed.endsWith("*/"))
+                {
+                    inJavadoc = false;
+                    matchNextLine = true;
+                }
+            }
+        }
+
+        Files.write(dest, destLines);
+
+        System.out.println("Copied Javadocs from " + generatedClass + " back to " + template);
+    }
+
+    private static void findLineReplacingJavadoc(List<String> lines, String matchLine, Deque<String> javadoc)
+    {
+        ListIterator<String> iter = lines.listIterator();
+        while (iter.hasNext())
+        {
+            String line = normalizeTemplateLine(iter.next().trim());
+            // Definitions aren't candidates for matches
+            if (line.isEmpty() || line.contains("::="))
+            {
+                continue;
+            }
+
+            if (line.equals(matchLine))
+            {
+                // Remove existing javadoc
+                iter.previous();
+                while (iter.hasPrevious()
+                    && (line = iter.previous().trim()).startsWith("@")
+                    || line.startsWith("*/")
+                    || line.startsWith("*")
+                    || line.startsWith("/**"))
+                {
+                    iter.remove();
+                }
+                iter.next();
+                javadoc.forEach(iter::add);
+                return;
+            }
+        }
+        System.err.println("Could not match line " + matchLine);
+        return;
+    }
+
+    // Remove all types and type placeholders so lines will exactly match
+    private static String normalizeTemplateLine(String s)
+    {
+        s = s.replace("<type1>", "<type>").replace("<type2>", "<type>");
+        s = s.replace("<name1>", "<name>").replace("<name2>", "<name>");
+        // boolean is a common return value for is...() methods - normalize
+        // it to match normalized lines from the generated class. Also remove \
+        return s.replace("boolean", "<type>").replaceAll(Pattern.quote("\\"), "");
+    }
+
+    // TODO: Multiple types/names not properly supported
+    private static String normalizeGeneratedClassLine(String s)
+    {
+        for (Primitive p : Primitive.values())
+        {
+            s = s.replace(p.type, "<type>").replace(p.getName(), "<name>");
+        }
+        // ... but "interface" gets erroneously replaced
+        return s.replace("<type>erface", "interface");
+    }
+
+    public static void main(String... args) throws IOException
+    {
+        if (args.length < 2 || args.length % 2 > 0)
+        {
+            throw new IllegalArgumentException("You must specify pairs of file paths: a generated class followed by its template");
+        }
+
+        for (int i = 0; i < args.length; i += 2)
+        {
+            new JavadocUtil().generatedClass(args[i]).template(args[i + 1]).process();
+        }
+    }
+}
```

Comment:
This utility is awesome! May be if it is possible, can you contribute another utility which will move code to the template? It will be super useful to be able to write code in one of the primitive files and reverse engineer it to a template. It need not be fully polished and working, but should provide a starting point. Thoughts?

Reply:
@nikhilnanivadekar I haven't forgotten about this request, but regretfully I'm not sure when I'll have the time to tackle it!